File: C++/ExtendedEuclidean/ExtendedEuclidean.cpp
#include <iostream>

int extendedEuclidean(int a, int b, int *x, int *y) {
	if (a == 0) {
		*x = 0;
		*y = 1;
		return b;
	}
	int x1, y1;
	int gcd = extendedEuclidean(b % a, a, &x1, &y1);
	*x = y1 - (b / a) * x1;
	*y = x1;
	return gcd;
}


int main() {
	int x, y, a, b;
	std::cout << "Enter the value of a, b: ";
	std::cin >> a >> b;
	int g = extendedEuclidean(a, b, &x, &y);

	std::cout << "The value of x, y is: " << x << ' ' << y << '\n';
}

File: algorithms/Java/ExtendedEuclidean/ExtendedEuclidean.java

class ExtendedEuclidean {

    /** This functions gives solution (x, y) of equation
     * ax + by = gcd(a, b)
     */
    static int[] extendedGcd(int a, int b) {
        // base case
        if (b == 0)
        {
            return new int[]{1, 0} ;
        }

        // recursive call
        int[] arr = extendedGcd(b, a%b) ;

        // calculating values of x and y for current call
        int x = arr[1] ;
        int y = arr[0] - (a/b)*(arr[1]) ;

        return new int[]{x, y} ;

    }

    public static void main(String[] args)
    {
        int a = 5, b = 8 ;
        int[] sol = extendedGcd(a, b) ;
        System.out.println("x= "+sol[0]+", y = "+ sol[1]);
    }

}
